src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -41,69 +46,108 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
+            var listOfValidParametersNodes = new List<SyntaxNode>();
+            var checkIfToReturn = false;
+
+            if (listOfParameterNodes != null)
             {
-                var parameterName = syntaxFacts.GetNameOfParameter(parameterNode);
-                if (parameterName == null || parameterName.Value.Span != context.Span)
+                foreach (var parameterNode in listOfParameterNodes)
+                {
+                    counter++;
+                    if (parameterNode == null)
+                    {
+                        continue;
+                    }
+
+                    // Only offered when there isn't a selection, or the selection exactly selects a parameter name.
+                    var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                    if (!context.Span.IsEmpty)
+                    {
+                        var parameterName = syntaxFacts.GetNameOfParameter(parameterNode);
+                        if (parameterName == null || parameterName.Value.Span != context.Span)
+                        {
+                            continue;
+                        }
+                    }
+
+                    var functionDeclaration = parameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
+                    if (functionDeclaration == null)
+                    {
+                        continue;
+                    }
+
+                    var parameterDefault = syntaxFacts.GetDefaultOfParameter(parameterNode);
+
+                    // Don't offer inside the "=initializer" of a parameter
+                    if (parameterDefault?.Span.Contains(position) == true)
+                    {
+                        continue;
+                    }
+
+                    var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                    // we can't just call GetDeclaredSymbol on functionDeclaration because it could an anonymous function,
+                    // so first we have to get the parameter symbol and then its containing method symbol
+                    var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
+                    if (parameter == null || parameter.Name == "")
+                    {
+                        continue;
+                    }
+
+                    var methodSymbol = (IMethodSymbol)parameter.ContainingSymbol;
+                    if (methodSymbol.IsAbstract ||
+                        methodSymbol.IsExtern ||
+                        methodSymbol.PartialImplementationPart != null ||
+                        methodSymbol.ContainingType.TypeKind == TypeKind.Interface)
+                    {
+                        continue;
+                    }
+
+                    if (CanOfferRefactoring(functionDeclaration, semanticModel, syntaxFacts, cancellationToken, out var blockStatementOpt))
```

Comment:
It seems almost everything in this if statement should be done after the loop on `listOfParameterNodes` (excepts possibly `listOfValidParametersNodes.Add(parameterNode);`)?